src/test/java/ReadingAggregatorTest.java

```diff
@@ -0,0 +1,125 @@
+import eventprocessing.fileservices.CSVFileService;
+import eventprocessing.models.Bucket;
+import eventprocessing.models.ReadingAggregator;
+import eventprocessing.models.SqsResponse;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.threeten.extra.MutableClock;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.temporal.ChronoUnit;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+public class ReadingAggregatorTest {
+
+    private ReadingAggregator readingAggregator;
+    private CSVFileService fileService;
+    private MutableClock clock;
+
+    @Before
+    public void buildReadingAggregator() {
+        clock = buildClock();
+        fileService = Mockito.mock(CSVFileService.class);
+        readingAggregator = new ReadingAggregator(fileService, clock);
+    }
+
+    @Test
+    public void process_nothingWhileNoReadingsAreOlderThanDelayTime() {
+        SqsResponse reading = buildReading(0);
+        readingAggregator.process(reading);
+        verify(fileService, never()).write(any());
+    }
+
+    @Test
+    public void process_singleReadingExceedingDelayTime_isSentToFileWriter() {
+        SqsResponse oldReading = buildReading(-5);
```

Comment:
I know the IDE shows the argument name, but the parameter is a bit opaque here. What about calling the method something like `readingWithMinutesDelay`?